File: backend/src/routes/hotspots.ts
import { Router } from "express";
import { getHotspots } from "../services/handlers.js";

export const hotspotsRouter = Router();

hotspotsRouter.get("/", async (req, res, next) => {
  try {
    const lat = Number(req.query.lat);
    const lng = Number(req.query.lng);
    const radius = Number(req.query.radius ?? 500);

    if (Number.isNaN(lat) || Number.isNaN(lng)) {
      res.status(400).json({ error: "lat and lng are required numbers" });
      return;
    }

    const data = await getHotspots(lat, lng, radius);
    res.json(data);
  } catch (e) {
    next(e);
  }
});

File: backend/guardian/routers/hotspots.py
from fastapi import APIRouter, HTTPException, Query

from guardian.services.handlers import get_hotspots

router = APIRouter()


@router.get("")
async def hotspots(
    lat: float = Query(...),
    lng: float = Query(...),
    radius: float = Query(400),
    limit: int = Query(800, ge=1, le=2000),
):
    if lat != lat or lng != lng:  # NaN check
        raise HTTPException(400, "lat and lng are required numbers")
    return await get_hotspots(lat, lng, radius, limit)
